client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java

```diff
@@ -405,6 +421,64 @@ public void testSessionProperties()
         }
     }
 
+    @Test(timeOut = 60_000)
+    public void testCancellationOnStatementClose()
+            throws Exception
+    {
+        String sql = "SELECT sleep_ms(10000) -- test cancellation " + randomTableSuffix();
+        Future<?> future;
+        try (Connection connection = createConnection()) {
+            try (Statement statement = connection.createStatement()) {
+                future = executor.submit(() -> statement.execute(sql));
+
+                // make sure query got executed
+                sleep(2_000);
+            }
+
+            // make sure query got cancelled by calling Statement::close
+            sleep(2_000);
```

Comment:
I think it would be better to use some sort of eventual assertion here rather than sleep with fixed amount.